src/Raven.Server/Documents/ETL/Providers/Raven/Enumerators/AttachmentTombstonesToRavenEtlItems.cs

```diff
@@ -0,0 +1,62 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using Raven.Server.ServerWide.Context;
+
+namespace Raven.Server.Documents.ETL.Providers.Raven.Enumerators
+{
+    public class AttachmentTombstonesToRavenEtlItems : IExtractEnumerator<RavenEtlItem>
+    {
+        private readonly DocumentsOperationContext _context;
+        private readonly IEnumerator<Tombstone> _tombstones;
+        private readonly List<string> _collections;
+
+        public AttachmentTombstonesToRavenEtlItems(DocumentsOperationContext context, IEnumerator<Tombstone> tombstones, List<string> collections)
+        {
+            _context = context;
+            _tombstones = tombstones;
+            _collections = collections ?? throw new ArgumentNullException(nameof(collections));
+        }
+
+        public bool Filter()
+        {
+            var tombstone = _tombstones.Current;
+            if (tombstone.Type != Tombstone.TombstoneType.Attachment)
+                throw new InvalidOperationException();
```

Comment:
Please provide exception message. Also you might extract the exception to a dedicated method to get slightly better perf. `Filter` is expected to be called quite frequently.

Reply:
And the exception message in other places as well.

Reply:
Frequent calls was my concern as well, but it is not suppose to be super performant anyway, right?
`Filter` will be called as much as `MoveNext`.

Reply:
Yeah. Just moving the `throw` clause to a separate method and calling it here might let it optimize stuff by inlining.